src/mdesktopentry.cpp

```diff
@@ -267,18 +266,19 @@ std::shared_ptr<QTranslator> MDesktopEntryPrivate::loadTranslator() const
 
     if (catalog.isEmpty() || !translator->load(QLocale(), catalog, "-", "/usr/share/translations")) {
         qDebug() << "Unable to load catalog" << catalog;
-        translator.reset();
+        delete translator;
+        translator = 0;
         translatorUnavailable = true;
     }
 
     if (translator) {
-        cachedTranslator = translator;
+        cachedTranslator.reset(translator);
         translatorCacheTimer.reset(new QTimer);
         translatorCacheTimer->setSingleShot(true);
         translatorCacheTimer->setInterval(TranslatorLifetime);
-        QObject::connect(translatorCacheTimer.data(), &QTimer::timeout, [this, translator] {
+        QObject::connect(translatorCacheTimer.data(), &QTimer::timeout, [this] {
+            cachedTranslator.take()->deleteLater();
```

Comment:
I guess I could just `cachedTranslator.reset()` but this felt safer...

Reply:
Should be fine to delete immediately, timeout shouldn't trigger while something it been translated with this.

Reply:
Changed that to reset, fixed one other small thing and squashed.